model/itemTab.cc

```diff
@@ -471,9 +491,21 @@ namespace minsky
                       itemCoords.erase(itemFocus);   
                       itemCoords.emplace(make_pair(itemFocus,make_pair(xItem,yItem)));         
                     } else cairo_translate(cairo,itemCoords[it].first,itemCoords[it].second);
-                    auto godley=GodleyTableWindow(g);
-                    godley.disableButtons();   
+                    GodleyTableWindow godley(g);
```

Comment:
Like my last comment - GodleyTableWindow is a little heavyweight for the draw method too.

My suggestion is to add another GodleyTableWindow member to GodleyIcon, like I did for the editor  mode

Reply:
Also lines 476 and 486 are a code smell that virtual function overrides should be used.

Reply:
> Like my last comment - GodleyTableWindow is a little heavyweight for the draw method too.
> 
> My suggestion is to add another GodleyTableWindow member to GodleyIcon, like I did for the editor mode

I tried this, but the best I get is when I declare a public "CopiableUniquePtr godley" in class GodlyeIcon, but which when I call it in ItemTab, returns a nullptr pointer and gives a segfault. 

I also tried initializing "CopiableUniquePtr godley" with *this in GodleyIcon, but no cigar, I'm afraid.

Then I tried adding AnotherCopiableUniquePtr with GodleyTableWindow as the base class, but this would not compile at all either.

Outside of simple types and classes, I really struggle with inheritance and polymorphism... :(

Reply:
> Also lines 476 and 486 are a code smell that virtual function overrides should be used.

What would be the return type of the overridden members in plotTab and godleyTab? ItemPtr?

Again, when the target function is of type void, bool, string, double, int, etc., I understand how to deal with virtual functions, but I don't know how to specialize to more general user defined classes or types.